src/HIndent.hs

```diff
@@ -211,73 +205,73 @@ test =
 testAst :: ByteString -> Either String (Module NodeInfo)
 testAst x =
   case parseModuleWithComments parseMode (UTF8.toString x) of
-    ParseOk (m,comments) ->
+    ParseOk (m, comments) ->
       Right
         (let ast =
                evalState
                  (collectAllComments
                     (fromMaybe m (applyFixities baseFixities m)))
                  comments
-         in ast)
+          in ast)
     ParseFailed _ e -> Left e
 
 -- | Default extensions.
 defaultExtensions :: [Extension]
 defaultExtensions =
-  [ e
-  | e@EnableExtension {} <- knownExtensions ] \\
+  [e | e@EnableExtension {} <- knownExtensions] \\
   map EnableExtension badExtensions
 
 -- | Extensions which steal too much syntax.
 badExtensions :: [KnownExtension]
 badExtensions =
-    [Arrows -- steals proc
-    ,TransformListComp -- steals the group keyword
-    ,XmlSyntax, RegularPatterns -- steals a-b
-    ,UnboxedTuples -- breaks (#) lens operator
+  [ Arrows -- steals proc
+  , TransformListComp -- steals the group keyword
+  , XmlSyntax
+  , RegularPatterns -- steals a-b
+  , UnboxedTuples -- breaks (#) lens operator
     -- ,QuasiQuotes -- breaks [x| ...], making whitespace free list comps break
-    ,PatternSynonyms -- steals the pattern keyword
-    ,RecursiveDo -- steals the rec keyword
-    ,DoRec -- same
-    ,TypeApplications -- since GHC 8 and haskell-src-exts-1.19
-    ]
-
+  , PatternSynonyms -- steals the pattern keyword
+  , RecursiveDo -- steals the rec keyword
+  , DoRec -- same
+  , TypeApplications -- since GHC 8 and haskell-src-exts-1.19
+  ]
 
 s8_stripPrefix :: ByteString -> ByteString -> Maybe ByteString
 s8_stripPrefix bs1@(S.PS _ _ l1) bs2
-   | bs1 `S.isPrefixOf` bs2 = Just (S.unsafeDrop l1 bs2)
-   | otherwise = Nothing
+  | bs1 `S.isPrefixOf` bs2 = Just (S.unsafeDrop l1 bs2)
+  | otherwise = Nothing
 
 --------------------------------------------------------------------------------
 -- Extensions stuff stolen from hlint
-
```

Comment:
Let's keep this empty line. The above comment is for an entire section in the file whereas the one below is just for the function

Reply:
Unfortunately, HIndent doesn't preserve empty lines between comments; fixing it requires modifying the source code, and I don't think it's an easy task.

Reply:
Hmm, let's open an issue for this too (if there isn't any)